app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/CreateScalingPolicyButton.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import autoBindMethods from 'class-autobind-decorator';
+
+import { Application, IServerGroup, ReactInjector } from '@spinnaker/core';
+
+import { PolicyTypeSelectionModal } from './upsert/PolicyTypeSelectionModal';
+import { UpsertTargetTrackingController } from './targetTracking/upsertTargetTracking.controller';
+import { IScalingPolicy } from 'amazon/domain';
+
+export interface ICreateScalingPolicyButtonProps {
+  application: Application;
+  serverGroup: IServerGroup;
+}
+
+export interface ICreateScalingPolicyButtonState {
+  showSelection: boolean;
+  showModal: boolean;
+  typeSelection: string;
+}
+
+@autoBindMethods
+export class CreateScalingPolicyButton extends React.Component<ICreateScalingPolicyButtonProps, ICreateScalingPolicyButtonState> {
+
+  constructor(props: ICreateScalingPolicyButtonProps) {
+    super(props);
+    this.state = {
+      showSelection: false,
+      showModal: false,
+      typeSelection: null,
+    };
+  }
+
+  public handleClick(): void {
+    this.setState({showSelection: true});
+  }
+
+  public createStepPolicy(): void {
+    const { serverGroup, application } = this.props;
+
+    const policy: IScalingPolicy = {
```

Comment:
These default values should probably live somewhere outside of the actual UI control itself, like how `constructNewClassicLoadBalancerTemplate` works.